### PR TITLE
Pagination Support for Job History

### LINKED ISSUE #4220: Support Pagination for fetching History from JobStore

### PR DESCRIPTION
**Pull request by udsamani**
This PR aims at the following

- Add support for pagination at JobStore layer for job histories

closes #4220

### CODE REVIEW

File: pkg/publicapi/endpoint/orchestrator/job.go [hunk 1]
```diff
@@ -92,7 +92,8 @@ func (e *Endpoint) getJob(c echo.Context) error { //nolint: gocyclo
 			if response.History != nil {
 				continue
 			}
-			history, err := e.store.GetJobHistory(ctx, jobID, jobstore.JobHistoryFilterOptions{})
+			jobHistoryQueryResponse, err := e.store.GetJobHistory(ctx, jobID, jobstore.JobHistoryQuery{})
+			history := jobHistoryQueryResponse.JobHistory
```
Comment (wdbaruni) [inline pkg/publicapi/endpoint/orchestrator/job.go:96]: you should do nil check validation first

**Review by wdbaruni — APPROVED** [review]
LGTM. Just a minor comment

File: pkg/publicapi/endpoint/orchestrator/job.go (hunk 1 above)
Comment (udsamani) [reply pkg/publicapi/endpoint/orchestrator/job.go:96]: Done.

Comment (frrist) [reply pkg/publicapi/endpoint/orchestrator/job.go:96]: I think we should check the error before assigning `history := jobHistoryQueryResponse.JobHistory` else this introduce the potential for panics if `GetJobHistory` returns `nil` and an `err`, which is a common practice.

**Pull request merged**